util/progress/metricwriter.go

```diff
@@ -446,14 +448,22 @@ func newLintMetricRecorder(meter metric.Meter, attrs attribute.Set) *lintMetricR
 	return mr
 }
 
+func kebabToCamel(s string) string {
+	words := strings.Split(s, "-")
+	for i, word := range words {
+		words[i] = cases.Title(language.English).String(word)
+	}
+	return strings.Join(words, "")
+}
+
 func (mr *lintMetricRecorder) Record(ss *client.SolveStatus) {
 	for _, warning := range ss.Warnings {
-		m := reLintMessage.FindSubmatch(warning.Short)
+		m := reLintMessage.FindSubmatch([]byte(warning.URL))
 		if m == nil {
 			continue
 		}
 
-		ruleName := string(m[1])
+		ruleName := kebabToCamel(string(m[1]))
```

Comment:
This should need a length check (already in old code)

Reply:
Do you mean a length check on the result from `FindSubmatch`?

Reply:
Yes, to avoid panic if `[]` is returned.